app/handlers/createHighlightVideo.py

```diff
@@ -11,12 +44,57 @@ class CreateHighlightVideoHandler(tornado.web.RequestHandler):
     @apiDescription Calling this route will create a highlight video of dangerous interactions from a specified project. When the video is created, an email will be sent to the project's user. This route requires running object tracking on the video, and then running safety analysis on the results of the object tracking beforehand. (Due to the potentially long duration, it is infeasible to return the results as a response to the HTTP request. In order to check the status of the testing and view results, see the Status group of messages.)
 
     @apiParam {String} identifier The identifier of the project to create a highlight video for.
-    @apiParam {Integer} [ttc_threshold] Threshold for determining whether an interaction is dangerous.
-    @apiParam {Integer} [vehicle_only] Flag for specifying only vehicle-vehicle interactions
+    @apiParam {Float} [ttc_threshold] Threshold for determining whether an interaction is dangerous. Default 1.5 seconds.
+    @apiParam {Integer} [vehicle_only] Flag for specifying only vehicle-vehicle interactions. Default to True.
 
     @apiSuccess status_code The API will return a status code of 200 upon success.
 
     @apiError error_message The error message to display.
     """
     def post(self):
-        self.finish("Create Highlight Video")
+        identifier = self.get_body_argument('identifier')
+        ttc_threshold = float(self.get_body_argument('ttc_threshold', default=1.5))
+        vehicle_only = bool(self.get_body_argument('vehicle_only', default=True))
+
+        status_code, reason = CreateHighlightVideoHandler.handler(identifier, ttc_threshold, vehicle_only)
+        if status_code == 200:
+            self.finish("Create Highlight Video")
+        else:
+            raise tornado.web.HTTPError(reason=reason, status_code=status_code)
+
+    @staticmethod
+    def handler(identifier, ttc_threshold, vehicle_only):
+        project_dir = get_project_path(identifier)
+        if not os.path.exists(project_dir):
+            return (500, 'Project directory does not exist. Check your identifier?')
+
+        db = os.path.join(project_dir, 'run', 'results.sqlite')
+        if not os.path.exists(db):
+            return (500, 'Database file does not exist. Trajectory analysis needs to be called first ')
```

Comment:
There should also be a check to see if tables like "interactions" exist.  Right now, this function will error if safety analysis was not called.